flow/Trace.cpp

```diff
@@ -53,6 +53,8 @@
 // 2. To avoid a historically documented but unknown crash that occurs when logging allocations
 //    during an open trace event
 thread_local int g_allocation_tracing_disabled = 1;
+thread_local unsigned tracedLines = 0;
```

Comment:
It looks like `tracedLines` is only used in `writeEvent` and is guarded by a mutex, so it doesn't need to be `thread_local`. It probably matters little given that it's used in simulation (though technically I think some simulated unit tests do actually use threads), but making it not `thread_local` would prevent the case where we have a lot of log lines spread over multiple threads.

```suggestion
unsigned tracedLines = 0;
```